sync3/lists.go

```diff
@@ -97,17 +97,8 @@ func (s *InternalRequestLists) SetRoom(r RoomConnMetadata) (delta RoomDelta) {
 		r.CanonicalisedName = strings.ToLower(
 			strings.Trim(internal.CalculateRoomName(&r.RoomMetadata, 5), "#!():_@"),
 		)
-		// Set LastInterestedEventTimestamps here so that recency sorts work. We use the
-		// timestamps provided by the caller, but otherwise fall back to that in the
-		// global event metadata.
-		newTimestamps := r.LastInterestedEventTimestamps
-		r.LastInterestedEventTimestamps = make(map[string]uint64, len(s.lists))
-		for listKey := range s.lists {
-			_, ok := newTimestamps[listKey]
-			if !ok {
-				newTimestamps[listKey] = r.LastMessageTimestamp
```

Comment:
I don't know why we're removing this. We haven't changed how the caller behaves, which is:
 - make a new map: `bumpTimestampInList := make(map[string]uint64, len(s.muxedReq.Lists))`
 - if there is a room update:
    * the list isn't using bump event types: set latest timestamp
    * the list is using bump event types: set only if event matches
 - call SetRoom with this map.

This code being deleted was useful as it preserved the previous values if they were not modified, whereas now they are being forgotten?

Reply:
AFAICS, The code that was deleted is plain wrong.

This takes place in the `else` branch (see -95 and -60) which is only called when we don't have a previous, already `existing` `*RoomConnMetadata`. I spy two wrongnesses:

1. `r.LastInterestedEventTimestamps` is set to an empty map on -104, but then _never written to afterwards_. 
2. `s.lists` is empty before `AssignList` is called for the first time. I believe the first `SetRoom` call happens before this(?)---if so, the loop on -105 does nothing.

By my reading, the old code completely dropped the LastInterestedTimestamps provided by the first SetRoom call.